core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -621,51 +615,17 @@ private int getLastItemEntryIndex() {
         return entryIndex;
     }
 
+
+    public void freeHandle(int handleIndex) {
+        handles[handleIndex].remove(memoryManager);
+    }
+
     /**
      * freezes chunk so no more changes can be done to it (marks pending items as frozen)
      */
     void freeze() {
         setState(State.FROZEN); // prevent new puts to this chunk
```

Comment:
should we change this to cas as well? for example if the chunk if released no need to freeze right?